src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnectionFactory.cs

```diff
@@ -86,23 +87,22 @@ internal static HttpConnectionOptions ShallowCopyHttpConnectionOptions(HttpConne
             {
                 HttpMessageHandlerFactory = options.HttpMessageHandlerFactory,
                 Headers = options.Headers,
-                Cookies = options.Cookies,
                 Url = options.Url,
                 Transports = options.Transports,
                 SkipNegotiation = options.SkipNegotiation,
                 AccessTokenProvider = options.AccessTokenProvider,
                 CloseTimeout = options.CloseTimeout,
-                Credentials = options.Credentials,
-                Proxy = options.Proxy,
-                UseDefaultCredentials = options.UseDefaultCredentials,
                 DefaultTransferFormat = options.DefaultTransferFormat,
                 WebSocketConfiguration = options.WebSocketConfiguration,
             };
 
-            // WASM doesn't support Crypto APIs and our setter throws if you try to assign null
-            if (options.ClientCertificates != null)
+            if (!OperatingSystem.IsBrowser())
             {
+                newOptions.Cookies = options.Cookies;
                 newOptions.ClientCertificates = options.ClientCertificates;
+                newOptions.Credentials = options.Credentials;
+                newOptions.Proxy = options.Proxy;
+                newOptions.UseDefaultCredentials = options.UseDefaultCredentials;
```

Comment:
Are all of these unsupported by HttpClient in the browser or just WebSockets?

Reply:
I just took a second look at the changes to HttpConnection make it so it no longer uses these properties when configuring the HttpClientHandler in the browser, so I hope we're sure these are unsupported.

But even if we're sure these are unnecessary, why go out of are way not to include them in the shallow copy?